composer.json

```diff
@@ -42,14 +42,18 @@
         "jetbrains/phpstorm-stubs": "^2019.1",
         "phpstan/phpstan": "^0.12",
         "phpunit/phpunit": "^8.4.1",
-        "symfony/console": "^2.0.5|^3.0|^4.0|^5.0"
+        "symfony/console": "^2.0.5|^3.0|^4.0|^5.0",
+        "vimeo/psalm": "^3.11"
     },
     "suggest": {
         "symfony/console": "For helpful console commands such as SQL execution and import of files."
     },
     "bin": ["bin/doctrine-dbal"],
     "config": {
-        "sort-packages": true
+        "sort-packages": true,
+        "platform": {
+            "php": "7.2.15"
```

Comment:
Should this be set to `7.2.0` (see https://github.com/doctrine/dbal/pull/3862)? Otherwise, we may end up updating the dependency constraints that do not support the lowest supported PHP version (which is 7.2.0).

Reply:
I originally used 7.2.0, and that did not work out because `symfony/console` requires 7.2.5, so I resolved to use whatever is used in the 7.2 Travis image we run the tests on.

Reply:
It means that technically the moment we bumped the `symfony/console` constraint, we dropped the support for the older PHP versions. If it's only the locked version that requires PHP 7.2.5, then I suggest we lock it to an older one that doesn't require PHP 7.2.5.

Reply:
Here is the message I obtained. I will try again, and will try to downgrade just the minimal amount of package to achieve this.
```
composer update nothing
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - symfony/console v5.0.2 requires php ^7.2.5 -> your PHP version (7.3.15) overridden by "config.platform.php" version (7.2.0) does not satisfy that requirement.
    - symfony/console v5.0.2 requires php ^7.2.5 -> your PHP version (7.3.15) overridden by "config.platform.php" version (7.2.0) does not satisfy that requirement.
    - symfony/console v5.0.2 requires php ^7.2.5 -> your PHP version (7.3.15) overridden by "config.platform.php" version (7.2.0) does not satisfy that requirement.
    - Installation request for symfony/console (locked at v5.0.2, required as ^2.0.5|^3.0|^4.0|^5.0) -> satisfiable by symfony/console[v5.0.2].
```

Reply:
Ok it was just 2 sf packages, will push shortly.

Reply:
The solution looks good.